packages/runtime-common/matrix-client.ts

```diff
@@ -96,6 +99,15 @@ export class MatrixClient {
     let json = await response.json();
 
     if (!response.ok) {
+      if (json['errcode'] === 'M_LIMIT_EXCEEDED') {
+        let retryAfter = json['retry_after_ms'];
+        this.#log.warn(
+          `Matrix rate limit exceeded, retrying after ${retryAfter}ms`,
+        );
+        await new Promise((res) => setTimeout(res, retryAfter));
+        return this.login();
```

Comment:
the HTTP API will be blocked on this too. this means that if this is triggered the HTTP API that the client is using will wait for 4 minutes. this seems not great....

Reply:
hmm ya, if it were possible to delay the job for the amount of time of the rate-limiting, would that make sense?

Reply:
well the thing is that we design our system so that you can read your own writes, meaning that the async for mutating an instance will not be resolved until the index has reflected the change that you made--so that you can read your own writes. I mean I get why you are doing this, we are basically screwed when we get this error code until `retry_after_ms` has elapsed. but it does mean that we'll be holding on to HTTP connections for a very ridiculously long time which can have bad side effects to other realms besides this one since the available connections to the server will drop. And in fact this could even be used as an attack vector to DDoS our realm server

Reply:
another reason why this is not great is that our queue is intentionally single threaded--we only want a single index mutation to be processed at once. We do not want concurrent index mutations. This means that if a worker is held up for 4 minutes, then all the subsequent index update jobs are also held up on this worker (and they may be for  totally different realms)